### ISSUE TITLE
[LifetimeSafety] Analysis does not report [[lifetimebound]] violation if annotated parameter escapes in an unrelated way

### ISSUE THREAD

**Issue by AbhinavPradeep** [llvm/llvm-project#200412]
The checking mechanism currently verifies [[lifetimebound]] for a parameter if it can find any related `OriginEscapesFact`. This causes false-negatives as below:

```c++
View G;

View assign_param_to_global(const MyObj &obj [[clang::lifetimebound]]) {
  G = obj;
  return View(); // unrelated view
}
```

Compiler explorer link: https://godbolt.org/z/4vqj6Ebcd

The above code should report that we `could not verify that the return value can be lifetime bound to 'obj'`, but currently the escape of `obj` to global `G` is taken as proof that it is lifetime bound. A similar example can be constructed for field escapes.
